pkg/resolver/resolver.go

```diff
@@ -402,6 +408,11 @@ func (r *resolver) handlePeerLookup(
 		Peers: cps,
 	}
 
+	if _, blacklisted := r.blacklist.Get(e.Sender.String()); blacklisted {
+		logger.Debug("address is blacklisted, skipping")
+		return
+	}
+
```

Comment:
You are checking if the peer who asked us is blacklisted.
Instead, when we find peers to respond with, we should not be sending blacklisted ones back.

We could maybe check the blacklist during `r.getClosest()`?

Reply:
We should be doing the same during `r.Lookup()` btw.